File: app/src/main/java/me/wangyuwei/xitukotlin/ui/MainActivity.java
package me.wangyuwei.xitukotlin.ui;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import me.wangyuwei.xitukotlin.R;
import me.wangyuwei.xitukotlin.entity.HotList;
import me.wangyuwei.xitukotlin.mvp.presenter.MainPresenter;
import me.wangyuwei.xitukotlin.mvp.views.MainView;
import me.wangyuwei.xitukotlin.ui.hot.HotAdapter;

/**
 * 作者： 巴掌 on 16/4/22 18:09
 */
public class MainActivity extends AppCompatActivity implements MainView{

    private RecyclerView rv_hot;
    private HotAdapter mHotAdapter;
    MainPresenter mainPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        rv_hot = (RecyclerView) this.findViewById(R.id.rv_hot);

        mHotAdapter = new HotAdapter(this);
        rv_hot.setAdapter(mHotAdapter);
        rv_hot.setLayoutManager(new LinearLayoutManager(this));

        mainPresenter = new MainPresenter(this);
        mainPresenter.loadData();

    }

    @Override
    public void showHotList(HotList hotList) {
        mHotAdapter.setData(hotList.getResults());
    }
}

File: app/src/main/java/me/wangyuwei/xitukotlin/ui/activity/MainActivity.kt
package me.wangyuwei.xitukotlin.ui.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView

import me.wangyuwei.xitukotlin.R
import me.wangyuwei.xitukotlin.entity.HotListEntity
import me.wangyuwei.xitukotlin.mvp.presenter.MainPresenter
import me.wangyuwei.xitukotlin.mvp.views.MainView
import me.wangyuwei.xitukotlin.ui.hot.HotAdapter
import me.wangyuwei.xitukotlin.widget.EndlessRecyclerOnScrollListener
import org.jetbrains.anko.find

/**
 * 作者： 巴掌 on 16/4/22 18:09
 */
class MainActivity : AppCompatActivity(), MainView{

    private val rv_hot: RecyclerView by lazy { find<RecyclerView>(R.id.rv_hot) }
    private lateinit var mHotAdapter: HotAdapter
    private lateinit var mainPresenter: MainPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mHotAdapter = HotAdapter(this)
        rv_hot.adapter = mHotAdapter
        val layoutManager: LinearLayoutManager = LinearLayoutManager(this)
        rv_hot.layoutManager = layoutManager
        rv_hot.addOnScrollListener(object : EndlessRecyclerOnScrollListener(layoutManager){
            override fun onLoadMore(currentPage: Int) {
                mainPresenter.loadMoreData(currentPage)
            }
        });

        mainPresenter = MainPresenter(this)
        mainPresenter.loadData()
    }

    override fun showHotList(hotList: HotListEntity) {
        mHotAdapter.setData(hotList.results)
    }

    override fun addHotListData(hotList: HotListEntity) {
        mHotAdapter.addData(hotList.results)
    }
}
